### PR TITLE
fix: enable process-migration when `tasklist` is enabled

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Always enable `process-migration` profile when `tasklist` is also enabled.
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: dist/src/main/java/io/camunda/application/commons/migration/MigrationProfilePostProcessor.java [hunk 1]
```diff
@@ -22,6 +22,9 @@ public void postProcessEnvironment(
     if (environment.acceptsProfiles(Profiles.of(Profile.MIGRATION.getId()))) {
       environment.addActiveProfile(Profile.PROCESS_MIGRATION.getId());
       environment.addActiveProfile(Profile.IDENTITY_MIGRATION.getId());
+    } else if (environment.acceptsProfiles(Profiles.of(Profile.TASKLIST.getId()))) {
```
Comment (ChrisKujawa) [inline dist/src/main/java/io/camunda/application/commons/migration/MigrationProfilePostProcessor.java:25]: :question: Can we be sure that the post processor is executed when no migration profil is set?

Comment (panagiotisgts) [reply dist/src/main/java/io/camunda/application/commons/migration/MigrationProfilePostProcessor.java:25]: Yes this post processor is defined in the [spring.factories](https://github.com/camunda/camunda/blob/d5dd4d107b02d354ff6c863363f001bcf402e0ea/dist/src/main/resources/META-INF/spring.factories) and it should be picked up always when a Spring application is running.

**Review by ChrisKujawa — APPROVED** [review]
Thanks :bow:

**Comment by panagiotisgts** [comment]
Hey @megglos, since the operations of the `process-migration` are idempotent we decided to just have it enabled by default when Tasklist is enabled, to ease the process of migrating to the CamundaExporter without having to specify additional profiles for just one time. It might need some additional fine tuning though. WDYT ?

**Pull request merged**